web/frontend/src/pages/election/components/DKGStatusRow.tsx

```diff
@@ -50,13 +62,107 @@ const DKGStatusRow: FC<DKGStatusRowProps> = ({
     signal: signal,
   };
 
+  const [showModal, setShowModal] = useState(false);
+
+  // Notify the parent each time our status changes
+  useEffect(() => {
+    notifyDKGState(node, InternalDKGInfo.fromStatus(status));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [status]);
+
+  // Notify the parent when we are loading or not
+  useEffect(() => {
+    notifyLoading(node, DKGLoading);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [DKGLoading]);
+
+  // send the initialization request
+  const initializeNode = async () => {
+    const req = {
+      method: 'POST',
+      body: JSON.stringify({
+        ElectionID: electionId,
+        Proxy: proxy,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    };
+
+    const response = await fetch(endpoints.dkgActors, req);
+    if (!response.ok) {
+      const txt = await response.text();
+      throw new Error(txt);
+    }
+  };
+
+  // Initialize the node if the initialization is ongoing and we are in a
+  // legitimate status.
+  useEffect(() => {
+    if (
+      ongoingAction === OngoingAction.Initializing &&
```

Comment:
Well it's actually the same problem as before, the parents are not notified since there are no changes the second time we click on Initializing (neither the Status nor the DKGLoading states change...)

Reply:
This should be solved in `Show.tsx` and inside the `notifyDKGState` function. There we can count the number of Initialized nodes (even if a node is already initialized, it will call this function at least once with its Initialized status), and update the status+ongoingAction once all nodes are initialized. What do you think ?

Reply:
I'm not sure I understood exactly what you meant
There's already a `useEffect` that checks if all the node were initialized and transitions to the Status Initialized if so (in `Show.tsx` line 165). But it's not triggered because none of the entries in `DKGStatuses` change. 
So yes if a node is already initialized, I think it should still call `notifyDKGState` so that it doesn't stay stuck in Initializing.

But I have no idea how it doesn't always remain stuck in Initializing because the `setOngoingAction(ongoingAction.None)` used to be handled in `useChangeAction` with the `onFulfilled/onReject` functions but is now missing for the case of the status Initialized... 😂

Reply:
I fixed that with the `DKGLoading`. Even if the status doesn't change, the node will always update its `DKGLoading` and therefore notify the parent with `notifyLoading`. The parent at that time knows that something changed on its children, and can make the necessary checks.

Reply:
Since we're allowing Unreachable nodes to still call the function, we need to make sure the proxy address at least could be retrieved (to avoid a request with an empty proxy address)

```suggestion
      ongoingAction === OngoingAction.Initializing && proxy !== '' &&
```

Reply:
@cmsigrist what about (added `if (proxy === '')`:

```ts
  useEffect(() => {
    if (
      ongoingAction === OngoingAction.Initializing &&
      (status === NodeStatus.NotInitialized ||
        status === NodeStatus.Failed ||
        status === NodeStatus.Unreachable)
    ) {
      setDKGLoading(true);

      if (proxy === '') {
        setDKGLoading(false);
        setInfo('proxy empty');
        return;
      }

      initializeNode()
        .then(() => {
          setInfo('');
          setStatus(NodeStatus.Initialized);
        })
        .catch((e: Error) => {
          setInfo(e.toString());
          setStatus(NodeStatus.Failed);
        })
        .finally(() => {
          setDKGLoading(false);
        });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [ongoingAction]);
```

Reply:
This works for the initialization (at least it passed the couple of failure cases I ran)  !  🎉

And then for the setup, I'm not sure if we're supposed to be able to retry setting up a node that has already failed the setup, but if this is the case, then it gets stuck in SettingUp.
It executes the `onrejected` part as expected (line 166):

```
      pollDKGStatus(expectedStatus)
        .then(
          () => {},
          (e: Error) => {
            setStatus(NodeStatus.Failed);
            setInfo(e.toString());
          }
        )
        .finally(() => setDKGLoading(false));
```
But it doesn't call `notifyDKGState`.

And then there's a minor bug where if we set up a first node and it fails but then set up another one which succeeds, the error message remains (the red little (!) next to the node status)

Reply:
> This works for the initialization (at least it passed all the failure cases I ran) ! 🎉

Yay 🥳

> I'm not sure if we're supposed to be able to retry setting up a node that has already failed the setup, but if this is the case

Yes, we should. 

> But it doesn't call notifyDKGState.

It should, when doing `setStatus(NodeStatus.Failed);`, then the `useEffect` in line 68 calls `notifyDKGState`, at least if the state was updated. Was it what you meant ?

Reply:
Yes it should, but I don't think it actually does, otherwise it wouldn't be stuck in Setting up ?

Reply:
True. I simplified some elements on `Show.ts` and it seems to work now ... 🥵